### PR TITLE
Ibp 3217 attributes brapi

### PR DESCRIPTION
**Pull request by vmaletta**
Please merge

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/domain/germplasm/AttributeDTO.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.generationcp.middleware.domain.germplasm;
+
+public class AttributeDTO {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/domain/germplasm/AttributeDTO.java]: implement toString, hashCode and equals

File: src/main/java/org/generationcp/middleware/dao/AttributeDAO.java [hunk 2]
```diff
@@ -88,4 +91,101 @@ public Attribute getAttribute(final Integer gid, final String attributeName) {
...
+			}
+
+			if (pageNumber != null && pageSize != null) {
+				query.setFirstResult(pageSize * (pageNumber - 1));
+				query.setMaxResults(pageSize);
+			}
+
+			final List<Object> results = query.list();
+
+			for (final Object o : results) {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/dao/AttributeDAO.java]: use ALIAS_TO_BEAN

File: src/main/java/org/generationcp/middleware/dao/AttributeDAO.java [hunk 3]
```diff
@@ -88,4 +91,91 @@ public Attribute getAttribute(final Integer gid, final String attributeName) {
 		}
 		return attribute;
 	}
+
+	public List<AttributeDTO> getAttributesByGidAndAttributeIds(
+		final String gid, final List<String> attributeIds, final Integer pageSize, final Integer pageNumber) {
+
+		List<AttributeDTO> attributes;
+		try {
+			String sql = "SELECT "
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/dao/AttributeDAO.java]: @vmaletta  You can make this a constant and do a select count(1) from the declared constant and we will have only one query to maintain

File: src/main/java/org/generationcp/middleware/domain/germplasm/AttributeDTO.java (hunk 1 above)
Comment (clarysabel) [reply src/main/java/org/generationcp/middleware/domain/germplasm/AttributeDTO.java]: @vmaletta Please add @AutoProperty annotation when using pojomatics

File: src/main/java/org/generationcp/middleware/manager/api/GermplasmDataManager.java [hunk 4]
```diff
@@ -1191,4 +1192,11 @@
 	long countGermplasmByStudy(Integer studyDbId);
 
 	List<GermplasmDTO> getGermplasmByStudy(Integer studyDbId, Integer pageNumber, Integer pageSize);
+
+	List<AttributeDTO> getAttributesByGid(
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/manager/api/GermplasmDataManager.java]: @vmaletta remove final from the parameters of the new functions in the interface

File: src/test/java/org/generationcp/middleware/dao/AttributeDAOTest.java [hunk 5]
```diff
@@ -71,18 +76,19 @@ private void setupTestData() {
...
 		this.testAttributeTypes = Arrays.asList(attributeType1, attributeType2, attributeType3);
-		
-		final Attribute attribute1 = new Attribute(null, germplasm1.getGid(), attributeType1.getFldno(), 1, RandomStringUtils.randomAlphabetic(100), null, null, null);
-		final Attribute attribute2 = new Attribute(null, germplasm1.getGid(), attributeType2.getFldno(), 1, RandomStringUtils.randomAlphabetic(100), null, null, null);
-		final Attribute attribute3 = new Attribute(null, germplasm2.getGid(), attributeType1.getFldno(), 1, RandomStringUtils.randomAlphabetic(100), null, null, null);
-		this.attributeDao.save(attribute1);
-		this.attributeDao.save(attribute2);
-		this.attributeDao.save(attribute3);
+
+		this.attribute1 = new Attribute(null, germplasm1.getGid(), attributeType1.getFldno(), 1, RandomStringUtils.randomAlphabetic(100), null, null, null);
```
Comment (clarysabel) [inline src/test/java/org/generationcp/middleware/dao/AttributeDAOTest.java]: @vmaletta use Random to generate the integer

File: src/main/java/org/generationcp/middleware/domain/germplasm/AttributeDTO.java (hunk 1 above)
Comment (vmaletta) [reply src/main/java/org/generationcp/middleware/domain/germplasm/AttributeDTO.java]: Ok

File: src/main/java/org/generationcp/middleware/dao/AttributeDAO.java (hunk 2 above)
Comment (vmaletta) [reply src/main/java/org/generationcp/middleware/dao/AttributeDAO.java]: Ready

File: src/main/java/org/generationcp/middleware/manager/api/GermplasmDataManager.java (hunk 4 above)
Comment (vmaletta) [reply src/main/java/org/generationcp/middleware/manager/api/GermplasmDataManager.java]: ok

File: src/test/java/org/generationcp/middleware/dao/AttributeDAOTest.java (hunk 5 above)
Comment (vmaletta) [reply src/test/java/org/generationcp/middleware/dao/AttributeDAOTest.java]: ok

**Pull request merged**